sky/utils/common_utils.py

```diff
@@ -182,7 +182,7 @@ def make_cluster_name_on_cloud(display_name: str,
             f'on the cloud, we convert it to {cluster_name_on_cloud}.')
     user_hash = ''
     if add_user_hash:
-        user_hash = get_user_hash()[:USER_HASH_LENGTH_IN_CLUSTER_NAME]
+        user_hash = get_user_hash()
```

Comment:
minor: we can further increase the character space for the user hash from hex to base36, i.e. update the `common_utils.generate_user_hash` to further encode the user_hash with `common_utils.base36_encode`.